connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/ExecutePlanHolder.scala

```diff
@@ -31,12 +32,16 @@ case class ExecutePlanHolder(
     "SparkConnect_" +
       s"User_${sessionHolder.userId}_Session_${sessionHolder.sessionId}_Request_${operationId}"
 
+  val events: RequestEvents = RequestEvents(this, new SystemClock())
+
   def interrupt(): Unit = {
     // TODO/WIP: This only interrupts active Spark jobs that are actively running.
     // This would then throw the error from ExecutePlan and terminate it.
     // But if the query is not running a Spark job, but executing code on Spark driver, this
     // would be a noop and the execution will keep running.
     sessionHolder.session.sparkContext.cancelJobsWithTag(jobTag)
-  }
 
+    // TODO: The above does not cancel the Connect query and it continues executing.
+    // events.postCanceled()
```

Comment:
can go to https://github.com/apache/spark/pull/41315/files#diff-cbfd25045877171d681719d5e689087697e3c6e8cf14772ba58242f535fdc11aR81 after that PR